url594c007908_hugobraga_College_tgz-pJ8-main_GraphJ8/src/main/Graph.java

```diff
@@ -0,0 +1,323 @@
+		String percCovTerFile = "percCovTer.dat";
+		String algoName = "ApproxAlgorithm-";
+		//String algoName = "SIM-";
+		
+		Date d = new Date();
+		String tempFile = "tempFile-"+d.getTime()+".dat";
+		Random random = new Random();    	
+    	
+    	
+    	int sourceId = Graph.SOURCE_ID;
+    	
+    	int terminalsId[];// = new int[Graph.NUMBER_OF_TERMINALS];
+    	double delaysTerm[];// = new double[Graph.NUMBER_OF_TERMINALS];
+    	
+    	double degree[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double violatedTerminalsCVR[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double minOfMeanViolatedTerminalsCVR[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double maxOfMeanViolatedTerminalsCVR[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double maxOfMaximumViolatedTerminalsCVR[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double meanOfMaximumViolatedTerminalsCVR[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	int violatedTerminalsCVRRUNSVector[][][] = new int[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double spCosts[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double terCoveredRatio[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	int terCoveredRatioRUNSVector[][][] = new int[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double violatedNodesRatio[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	int violatedNodesRatioRUNSVector[][][] = new int[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double violatedRunsRatio[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	double violatedRunsRatioRUNSVector[][][] = new double[netSizes.length][terminalSizes.length][Graph.SPANNER_FACTORS.length];
+    	
+    	double degreeSPT[][] = new double[netSizes.length][terminalSizes.length];
+    	double costsSPT[][] = new double[netSizes.length][terminalSizes.length];    	
+		
+		try {
+			
+			PrintWriter degreeOut = new PrintWriter(new BufferedWriter(new FileWriter(dir+algoName+outDegreeFile)));
+			PrintWriter violatedTerminalsCVRFileOut = new PrintWriter(new BufferedWriter(new FileWriter(dir+algoName+violatedTerminalsCVRFile)));
+			PrintWriter violatedNodesRatioOut = new PrintWriter(new BufferedWriter(new FileWriter(dir+algoName+violatedNodesRatioFile)));
+			PrintWriter violatedRunsRatioOut = new PrintWriter(new BufferedWriter(new FileWriter(dir+algoName+violatedRunsRatioFile)));
+			PrintWriter percCovTerOut = new PrintWriter(new BufferedWriter(new FileWriter(dir+algoName+percCovTerFile)));			
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *percCovTerOut*. The resource is referenced at lines 103, 240, 253. The resource is closed at line 261. There are other execution paths that do not contain closure statements, for example, when FileWriter constructor throws an exception. Ensure *percCovTerOut* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).